### DISCUSSION TITLE
What is eating up my datasore?!

### DISCUSSION THREAD

**Discussion by dh9dat** [minio/minio#18462] [Q&A]
Hello all,

I hope this is the right place for this kind of questions, if not, please tell me what's the correct way... I have setup MinIO on a Synology storage and created a bucket which should be used as capacity tier for a Veeam backup system.
MinIO shows a data capacity of 163TB, and a usage of 161 TB, but the bucket (which is the only one present) has only 4TB in 8.5M items:
![Screenshot from 2023-11-16 14-34-13](https://github.com/minio/minio/assets/56773213/797d6323-7a8b-49b8-81f3-0174d3703be7)
So, can someone tell me what's filling up the datastorage? Versioning or similar is not activated for the bucket.

Thanks and best regards,

Harald

**Comment by klauspost** [comment]
Are you sharing the `/data` device with other parts of your FS? Please describe your setup.

**Comment by dh9dat** [comment]
Hello Klaus,

thanks for your response and please excuse my real late reply: I had already posted an answer, but it seems that my browser has eaten it up, or it didn't made it to the server farm anyhow... And, I was bit busy in the last weeks, so, I forgot to check it, sorry.
Ok, so, let me re-post it and explain my setup: As said, I've installed MinIO on a Synology storage which is used exclusively for this, with no other data. The installation was done by using this HowTo: https://www.jonaharagon.com/posts/installing-minio-on-synology-diskstation/ (thanks to Jonah Aragon for posting it).
As described therein, I've created a volume which is spanned over the entire space (163TB), and two shared folders: One for the data, and another one for the Docker containers. After installing and setting up MinIO inside Docker I've installed Synology's "Web station" plugin to get external access to the MinIO UI, this created two additional shared folders (web and web_packages).
I've checked the file shares, there's 2,9kB in docker, abt. 2kB in web and nothing in web_packages. So, the only share with a significant content is data.

Thanks and best regards from Germany,

Harald

**Reply by cgs-labs** [reply]
So you've got a reported usage of 4TiB, but you use 161.3TiB on disk.

Minio does not delete files. It rather marks then as deleted, but keeps them on disk. Have a look at your bucket and check the "show deleted objects" box. The deleted objects appear in gray.
![minio](https://github.com/minio/minio/assets/65296245/3943eff3-771e-4f2d-b355-3c0b96758635)

If you did not expect this, or if you really want the delete the files, the easiest way is to create a lifecycle management rule.
You can expire non-current objects after some time, say 1 day or so. But be careful when you use versioning for other reasons.

73, Chris

**Comment by dh9dat** [comment]
Hello Chris,

thanks for your answer, I've checked my bucket but I don't see this checkbox?
![image](https://github.com/minio/minio/assets/56773213/addc2e5c-1a93-4e40-9818-90d209151ed5)
Nevertheless, I've now created a lifecycle rule which should remove all deleted files after one day (that's okay since Veeam has its own lifecyle policy).
Is it possible to run this rule manually to clean up the data storage? If not, I'll just wait until Monday and check the data storage again.

Thanks and best 73's from Germany,

Harald

**Reply by harshavardhana** [reply]
mc rm --non-current --versions -r

**Comment by dh9dat** [comment]
Hello harshavardhana,

thanks for your quick reply! It took me a moment to get access to the bucket since I'm not so familiar with MinIO and Containers, but I finally got it. So, I've issued the command (in dry-run mode), but it seems that it takes a pretty amount of time (which is okay since the bucket is really filled up with several little files).
I'll let it run and see how long it takes, I'll post an update when there are any news.

Thanks and best regards from Germany,

Harald

**Comment by dh9dat** [comment]
Hello harshavardhana,

I've looked at the storage, and the command was still running after two days, so, I've canceled it. Could be that it is taking so long because Veeam still tries to write data to the storage...
I will now set the storage to maintenance mode in Veeam, and take a look if my lifecycle rule which I've also configured removes the older versions. If yes, I should see some more free space tomorrow.

Thanks again for your support and best regards from Germany,

Harald
